### ISSUE TITLE
Underestimation of the right frontal GM

### ISSUE THREAD

**Issue by rcruces** [Deep-MI/FastSurfer#351]
## Question/Support Request
There is a consisten underestimation of the GM surface of the right frontal lobe in various subjects of a database (MICs https://portal.conp.ca/dataset?id=projects/mica-mics).  The issue appear to be happening during the white matter surface generation and is independent of the reconstruction algorithm (fsqsphere, spectral spherical projection). We also tried three different versions (cpu-v2.0.2, cpu-v2.0.4 and cpu-v2.1.1) all of them with a singularity container build from the dockerhub and different pre processing of the T1w images (rawimage, biasfiled corrected, non local mean denoising, bias field corrected+non local means). And different subjects, however the results were always similar.

The following files have a very accurate estimation of the GM:
- mask
- aparc.DKTatlas+aseg.orig
- aparc.DKTatlas+aseg.deep.withCC

Problem begin on `wm.seg.mgz`  where some voxels are missing from the `aparc.DKTatlas+aseg.deep` . Then `ribbon.mgz` already contains the missing information. As well as all the following files:
- aparc.DKTatlas+aseg
- aparc.DKTatlas+aseg.mapped
- aseg.mgz

## Screenshots
Each image correspond a different subject. The issue is always on the right frontal lobe and is not relate to a bias field artifact or difference in the intensity values of the GM/WM.

![Screen Shot 2023-06-19 at 1 49 24 PM](https://github.com/Deep-MI/FastSurfer/assets/15281416/3aff8193-665e-4066-a22c-4990599aedf3)

![Screen Shot 2023-06-20 at 2 12 26 PM](https://github.com/Deep-MI/FastSurfer/assets/15281416/43e3331d-5c01-4036-9c17-abb1abd4ef02)

![Screen Shot 2023-06-20 at 2 13 33 PM](https://github.com/Deep-MI/FastSurfer/assets/15281416/3c5ee3ec-4f63-4ad1-aa85-54d31b1727)
![Screen Shot 2023-06-22 at 10 37 57 AM](https://github.com/Deep-MI/FastSurfer/assets/15281416/37c4cedc-6623-447f-9807-ca597ce1f353)
6e)
![screenshot_2023-06-19_14-13-55](https://github.com/Deep-MI/FastSurfer/assets/15281416/eb45f98a-e326-49db-8701-9936087bbd24)
![Screen Shot 2023-06-20 at 1 47 41 PM](https://github.com/Deep-MI/FastSurfer/assets/15281416/165a0122-512a-4b1f-9e39-6e59e5330d48)

## Environment
 - FastSurfer Version: cpu-v2.0.2, cpu-v2.0.4 and cpu-v2.1.1 docker/singularity image

### Execution
Two additional errors were found while running the container function `run_fastsurfer.sh` due to aproblem reading the flag `--threads`. 

`deepmi/fastsurfer      cpu-v2.0.4`
```
./recon-surf.sh --sid v2.0.4_sub-HC071_ses-01_T1w_N4 --sd /output --t1 /output/v2.0.4_sub-HC071_ses-01_T1w_N4/mri/orig.mgz --aparc_aseg_segfile /output/v2.0.4_sub-HC071_ses-01_T1w_N4/mri/aparc.DKTatlas+aseg.deep.mgz --parallel --threads --py python3.8
ERROR: Flag python3.8 unrecognized.
```

`deepmi/fastsurfer      cpu-v2.1.1`
```
python3.8 /fastsurfer/recon_surf/N4_bias_correct.py --in /output/v2.1.1_sub-HC071_ses-01_T1w/mri/orig.mgz --out /output/v2.1.1_sub-HC071_ses-01_T1w/mri/orig_nu.mgz --mask /output/v2.1.1_sub-HC071_ses-01_T1w/mri/mask.mgz --threads
ERROR: Biasfield correction failed
```

This issue was solved by running the commands separately:
`fastsurfer_img` correspond to each of the following: `cpu-v2.0.2`, `cpu-v2.0.4` and `cpu-v2.1.1`
```
singularity exec --nv -B ${TMPDIR}:/${TMPDIR} \
-B "${SUBJECTS_DIR}":/output -B "${T1}/":/anat/T1w.nii.gz \
-B "${fs_licence}:/license.txt" 
"${fastsurfer_img}" /fastsurfer/run_fastsurfer.sh --fs_license /license.txt \
--t1 /anat/T1w.nii.gz --sid "${idBIDS}" --sd /output --parallel --seg_only --threads 15

rm -v ${SUBJECTS_DIR}/${idBIDS}/mri/aparc.DKTatlas+aseg.orig.mgz

singularity exec --nv -B ${TMPDIR}:/${TMPDIR} \
-B "${SUBJECTS_DIR}":/output  \
-B "${fs_licence}:/license.txt" \
"${fastsurfer_img}" /fastsurfer/recon_surf/recon-surf.sh --fs_license /license.txt \
--t1 /output/${idBIDS}/mri/orig.mgz --sid "${idBIDS}" --sd /output --parallel --threads 15
```

We really thank you for any information or support regarding this issue!

**Comment by m-reuter** [comment]
Hi @rcruces , 
thanks for reporting. For the surface placement we run FreeSurfer code. Have you tried running FreeSurfer to see how that behaves? It looks to me like our segmentation pipeline (FastSurferVINN) does a pretty good job, but then later the surface is shifted too far inside. This could be caused by a bad bias correction and too dark GM intensity values in those areas. I am not a big fan of the N4 (ITK) bias correction. Also you had an error during that step. Usually there should not be any such errors and it could indicate a quality issue with the images. Also the python3.8 flag error should not be there. 

So first we would like to see if there are any bugs with the specific flags you used. For that can you post the command(s) that lead to those errors?

In the meantime, you can try fixing the (potential) intensity issue by:
- comparing to the output of FreeSurfer (which probably shows similar results)
- switch on the old N3 bias field correction (in FreeSurfer), to see if that fixes it. We could then add a flag for that also in FastSurfer.

**Comment by m-reuter** [comment]
Hi again, 

I ran Freesurfer 7.4.1 default stream and also using N3 instead of the N4 bias correction. Both create similar surfaces that are even tighter than Fastsurfer's pial surface in most places (in the one case I looked at from that data set). 

GM is very dark in those rh frontal regions. In most other areas it is between 60-70 in FastSurfer's norm.mgz or even down to 50, but in the frontal areas it is below 50 which probably causes surfaces to stop expanding there. 

I am not sure what causes intensities to be so low there or how to fix it (maybe a more time consuming SPM bias correction as a preprocessing step can help here).

We should note that this is not a special FastSurfer issue, but that FreeSurfer behaves similarly if not worse. 

Finally, I ran FastSurfer with --threats 2 and with --parallel and had no issues. In order for us to investigate further, we need the exact command that you used that led to those error. Make sure to provide an integer number after --threats N to specify the number of threats to use. In your command above that is missing and so it might eat the --py flag and then does not know what the "python3.8" is supposed to mean.

**Comment by m-reuter** [comment]
Note that we fixed some things related to the bias field correction in #387 and #406 which have been released in V2.2.0 last December. It would be interesting to know if maybe those fixes help with this data also? It could affect surface placement.

**Comment by m-reuter** [comment]
closing this for now, feel free to reopen if the issue still persists.
